### PR TITLE
241 ps/expimpsolver improvements

### LINKED ISSUE #241: Improve Explicit/Implicit Demonstrator
Gerd had some comments

### PR DESCRIPTION
**Pull request by PhiSpel**
# Description

Gerd had some comments about the Explicit/Implicit comparison demonstrator.
Closes #241 

## Checklist

 - [x] This pull request is associated to an issue
 - [x] All used resources use a CC-by-SA license or less and are marked appropriately, see [our contribution guide](CONTRIBUTING.md).
 - [x] All used resources have been added to the list in `[TODO, does not exist yet]`
 - [x] New exercises include learning requirements and learning goals.
 - [x] Add Lea as reviewer

### CODE REVIEW

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
+    "# Stabilität des expliziten versus impliziten Eulerverfahrens\n",
     "\n",
-    "Bestimmte ODEs lassen sich mit impliziten Verfahren besser lösen als mit expliziten Verfahren. Diese ODEs bezeichnen wir als **steif**.\n",
+    "Bestimmte ODEs lassen sich mit impliziten Verfahren besser lösen als mit expliziten Verfahren. Diese ODEs bezeichnen wir als **steif**. Das möchten wir an einem kleinen Modellproblem demonstrieren.\n",
     "\n",
-    "Das liegt vor allem an der hohen Lipschitzkonstante $\\lambda$. Diese beschreibt den maximalen Betrag der Steigung. Also:\n",
+    "Entscheidend dafür, ob eine hohe maximale Steigung vorliegt. Diese wird mit der Lipschitzkonstanten beschrieben, die eine obere Schranke der Steigung $f_y$ aufstellt (mit der ODE $y' = f(t,y)$):\n",
     "\n",
-    "$$\\lambda \\ge \\left|\\frac{f(x_1)-f(x_2)|}{x_1-x_2}\\right| \\,\\forall\\, x_1,x_2 \\in \\mathbb{D}$$\n",
+    "$$\\lambda \\ge \\left|\\frac{f(t,y_1)-f(t,y_2)|}{y_1-y_2}\\right| \\,\\forall\\, y_1,y_2 \\in \\mathbb{D}$$\n",
```
Comment (m-r-schenk) [inline content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: I learned, that the universal quantifier comes first, before the equation.

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
    "metadata": {},
    "source": [
-    "# Stabilität Eulerverfahren\n",
+    "# Stabilität des expliziten versus impliziten Eulerverfahrens\n",
     "\n",
-    "Bestimmte ODEs lassen sich mit impliziten Verfahren besser lösen als mit expliziten Verfahren. Diese ODEs bezeichnen wir als **steif**.\n",
+    "Bestimmte ODEs lassen sich mit impliziten Verfahren besser lösen als mit expliziten Verfahren. Diese ODEs bezeichnen wir als **steif**. Das möchten wir an einem kleinen Modellproblem demonstrieren.\n",
     "\n",
-    "Das liegt vor allem an der hohen Lipschitzkonstante $\\lambda$. Diese beschreibt den maximalen Betrag der Steigung. Also:\n",
+    "Entscheidend dafür, ob eine hohe maximale Steigung vorliegt. Diese wird mit der Lipschitzkonstanten beschrieben, die eine obere Schranke der Steigung $f_y$ aufstellt (mit der ODE $y' = f(t,y)$):\n",
```
Comment (m-r-schenk) [inline content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb:11]: This is not a proper sentence...

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb [hunk 3]
```diff
@@ -4,15 +4,15 @@
...
     "\n",
-    "$$\\lambda \\ge \\left|\\frac{f(x_1)-f(x_2)|}{x_1-x_2}\\right| \\,\\forall\\, x_1,x_2 \\in \\mathbb{D}$$\n",
+    "$$\\forall y_1,y_2 \\in \\mathbb{D} \\left(\\lambda \\ge \\left|\\frac{f(t,y_1)-f(t,y_2)|}{y_1-y_2}\\right|\\right)$$\n",
```
Comment (m-r-schenk) [inline content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb:13]: I learned that the universal quantifier comes first, before the formula...

**Review by m-r-schenk — CHANGES_REQUESTED** [review]

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 2 above)
Comment (LeaKruehler) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: I changed it to "Entscheidend dafür ist, ob eine hohe maximale Steigung vorliegt."

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 3 above)
Comment (LeaKruehler) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: I didn't change that, because I have no clue about what you mean.

**Comment by LeaKruehler** [comment]
Is ready to be merged after Martins requested changes about the universal quantifier and formula have been made.

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 2 above)
Comment (PhiSpel) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: This works, sorry, I think I just forgot the "ist"

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 3 above)
Comment (PhiSpel) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: I have no opinion about this, I just know it from differentiating between cases, i.e. $|x|=\left{\begin{array} x & \forall x \ge 0 \\ -x & \forall x < 0 \right$

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb [hunk 4]
```diff
@@ -258,7 +258,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "**MATLAB Code**"
+    "**Matlab Code**"
```
Comment (PhiSpel) [inline content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb:261]: Mathworks spells MATLAB in all-caps. I agree on unifying this, we can just briefly agree on which spelling to use in the next meeting

Comment (PhiSpel) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb:261]: Will be spelled `Matlab`

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 3 above)
Comment (PhiSpel) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: Works like this

File: content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb (hunk 1 above)
Comment (PhiSpel) [reply content/07_num_ode/02_stabilitaet_eulerverfahren.ipynb]: ok

**Comment by PhiSpel** [comment]
depends on #206

**Review by m-r-schenk — APPROVED** [review]

**Pull request merged**